File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/enrollment/event/EnrollmentAccepted.java
package cloud.ambar.creditcard.enrollment.event;

import cloud.ambar.common.event.TransformationEvent;
import cloud.ambar.creditcard.enrollment.aggregate.Enrollment;
import cloud.ambar.creditcard.enrollment.aggregate.EnrollmentStatus;
import lombok.Getter;
import lombok.NonNull;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@Getter
public class EnrollmentAccepted extends TransformationEvent<Enrollment> {
    @NonNull private String reasonCode;
    @NonNull private String reasonDescription;

    public Enrollment transformAggregate(Enrollment aggregate) {
        return aggregate.toBuilder()
                .aggregateVersion(aggregateVersion)
                .status(EnrollmentStatus.ACCEPTED.name())
                .build();
    }
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Enrollment/Event/EnrollmentAccepted.cs
using CreditCardEnrollment.Common.Event;

namespace CreditCardEnrollment.CreditCard.Enrollment.Event;

public class EnrollmentAccepted : TransformationEvent<Aggregate.Enrollment>
{
    public required string ReasonCode { get; init; }
    public required string ReasonDescription { get; init; }

    public override Aggregate.Enrollment TransformAggregate(Aggregate.Enrollment aggregate)
    {
        return new Aggregate.Enrollment
        {
            AggregateId = AggregateId,
            AggregateVersion = AggregateVersion,
            UserId = aggregate.UserId,
            ProductId = aggregate.ProductId,
            Status = Aggregate.EnrollmentStatus.Accepted,
            AnnualIncomeInCents = aggregate.AnnualIncomeInCents,
            EnrollmentFirstRequestedOn = aggregate.EnrollmentFirstRequestedOn
        };
    }
}
